cmd/storage-rest-client.go

```diff
@@ -210,18 +239,29 @@ func (client *storageRESTClient) SetDiskID(id string) {
 
 // DiskInfo - fetch disk information for a remote disk.
 func (client *storageRESTClient) DiskInfo(ctx context.Context) (info DiskInfo, err error) {
-	respBody, err := client.call(ctx, storageRESTMethodDiskInfo, nil, nil, -1)
-	if err != nil {
-		return info, err
-	}
-	defer http.DrainBody(respBody)
-	if err = msgp.Decode(respBody, &info); err != nil {
-		return info, err
-	}
-	if info.Error != "" {
-		return info, toStorageErr(errors.New(info.Error))
+	client.diskInfoCache.Once.Do(func() {
+		client.diskInfoCache.TTL = time.Second
+		client.diskInfoCache.Update = func() (interface{}, error) {
+			respBody, err := client.call(context.Background(), storageRESTMethodDiskInfo, nil, nil, -1)
+			if err != nil {
+				return info, err
+			}
+			defer http.DrainBody(respBody)
+			if err = msgp.Decode(respBody, &info); err != nil {
+				return info, err
+			}
+			if info.Error != "" {
+				return info, toStorageErr(errors.New(info.Error))
+			}
+			return info, nil
+		}
+	})
+	val, err := client.diskInfoCache.Get()
+	if err == nil {
+		info = val.(DiskInfo)
 	}
-	return info, nil
+
+	return info, err
```

Comment:
This was removed on purpose @klauspost never hold write locks while making network calls - because a potential hang in-network can hang other calls to DiskInfo() and serialize them. 

So it is on purpose removed to avoid such serializations.  This was observed at a customer setup where HTTP2 was hung and it hung all DiskInfo calls.

Reply:
This is called so frequently by `getLoadBalancedDisks`, `getOnlineDisks`, `getOnlineDisksWithHealing` it is virtualy being called once per operation. The chattyness of that is crazy.